### PR TITLE
[RFC] tests: Migrate legacy test 79.

### PR DESCRIPTION
**Pull request by lucc**

### CODE REVIEW

File: test/functional/legacy/079_substitute_special_spec.lua [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+local eq, eval, write_file = helpers.eq, helpers.eval, helpers.write_file
+
+local function expect_line(line)
+  return eq(line, eval('getline(".")'))
+end
+
+describe(':substitue', function()
+  setup(function()
+    write_file('test_79_7.in', 'TEST_7:\nA\rA\nB\x00B\nC\x00C\nD\x00\nD\nE'..
+      '\x00\n\x00\n\x00\n\x00\n\x00E\nQ\nQ\n')
```
Comment (Pyrohh) [inline test/functional/legacy/079_substitute_special_spec.lua]: must this be written to a file, as opposed to just being `insert()`ed into a buffer?

**Comment by Pyrohh** [comment]
Just a heads up, please add any changes from review as separate commits. I find it makes reviews much easier, and I can just squash the commits immediately prior to merging.

File: test/functional/legacy/079_substitute_special_spec.lua [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+  end)
+  before_each(clear)
+  teardown(function()
+    os.remove('test_79_7.in')
+  end)
+
+  it('with "set magic"', function()
+    insert([[
+      TEST_1:
```
Comment (Pyrohh) [inline test/functional/legacy/079_substitute_special_spec.lua]: perhaps the same thing that was done in test80 could be done here, where the `TEST_N` text could be moved into the first argument of `it()`.

File: test/functional/legacy/079_substitute_special_spec.lua [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+      mMm
+      nNn
+      oOo
+      pPp
+      qQq
+      rRr
+      sSs
+      tTt
+      U
+      V
```
Comment (Pyrohh) [inline test/functional/legacy/079_substitute_special_spec.lua]: I imagine that most of this test could be converted to use `eq()`, but only if you want to (I'm guessing it's a lot of extra work).

File: test/functional/legacy/079_substitute_special_spec.lua (hunk 1 above)
Comment (lucc) [inline test/functional/legacy/079_substitute_special_spec.lua]: It is not necessary but the null byte has to be escaped with <C-V> in insert(). I thought it is a hassle and easier to read like this.

File: test/functional/legacy/079_substitute_special_spec.lua (hunk 3 above)
Comment (lucc) [inline test/functional/legacy/079_substitute_special_spec.lua]: You are right it would be closer to the new tests. But I think these tests are very simple and quite readable and easy to follow like this.

File: test/functional/legacy/079_substitute_special_spec.lua (hunk 2 above)
Comment (lucc) [inline test/functional/legacy/079_substitute_special_spec.lua]: done

**Comment by lucc** [comment]
ref: part of #1286.

**Comment by justinmk** [comment]
addressed by 2de447b60c04. In some cases it's still worth merging in the Lua version, but for cases like this where it's hard to follow what's going on, it's risky to maintain an extra test.
